{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pymongo.results.InsertManyResult at 0x7f9e2289a070>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# import pymongo\n",
    "# # Read the CSV file into a dataframe\n",
    "\n",
    "# # Connect to the MongoDB server\n",
    "# client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\")\n",
    "\n",
    "# # Select the database and collection\n",
    "# db = client[\"data-management-db\"]\n",
    "# collection = db[\"google_data\"]\n",
    "\n",
    "# # Insert the dataframe into the collection\n",
    "# collection.insert_many(df.to_dict('records'))\n",
    "import pymongo\n",
    "df = pd.read_csv(\"./output_files/google_dataset_cleaned_2.csv\")\n",
    "df2 = pd.read_csv(\"./output_files/trip_advisor_cleaned.csv\")\n",
    "\n",
    "# # 1. Verify that the MongoDB server is running.\n",
    "# try:\n",
    "#     client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\", serverSelectionTimeoutMS=1000)\n",
    "#     client.server_info()\n",
    "#     print(\"MongoDB server is running.\")\n",
    "# except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "#     print(\"Error: MongoDB server is not running.\")\n",
    "\n",
    "# # 2. Verify that the MongoDB server is reachable at the host and port specified in the connection string\n",
    "# try:\n",
    "#     client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\", serverSelectionTimeoutMS=1000)\n",
    "#     client.server_info()\n",
    "#     print(\"MongoDB server is reachable at localhost:27017.\")\n",
    "# except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "#     print(\"Error: MongoDB server is not reachable at localhost:27017.\")\n",
    "\n",
    "# # 3. Check if there's a firewall or a security group blocking the connection to the MongoDB server\n",
    "# try:\n",
    "#     client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\", serverSelectionTimeoutMS=1000)\n",
    "#     client.server_info()\n",
    "#     print(\"Connection to MongoDB server is successful.\")\n",
    "# except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "#     print(\"Error: Connection to MongoDB server is blocked by firewall or security group.\")\n",
    "\n",
    "# # 4. Check if the hostname is correct and the DNS resolution is working as expected\n",
    "# try:\n",
    "#     client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\", serverSelectionTimeoutMS=1000)\n",
    "#     client.server_info()\n",
    "#     print(\"DNS resolution is working and hostname is correct.\")\n",
    "# except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "#     print(\"Error: DNS resolution is not working or hostname is incorrect.\")\n",
    "\n",
    "# # 5. Verify that the MongoDB server is configured to listen on the IP address and port that you are trying to connect to\n",
    "# try:\n",
    "#     client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@mongo:27017/\", serverSelectionTimeoutMS=1000)\n",
    "#     client.server_info()\n",
    "#     print(\"MongoDB server is configured to listen on localhost:27017.\")\n",
    "# except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "#     print(\"Error: MongoDB server is not configured to listen on localhost:27017.\")\n",
    "\n",
    "# # 6. Try using the IP address '0.0.0.0'\n",
    "try:\n",
    "    client = pymongo.MongoClient(\"mongodb://admin:DataMan2022!@0.0.0.0:27017/\", serverSelectionTimeoutMS=1000)\n",
    "    client.server_info()\n",
    "    print(\"MongoDB server is reachable using IP address 0.0.0.0.\")\n",
    "except pymongo.errors.ServerSelectionTimeoutError as err:\n",
    "    print(\"Error: MongoDB server is not reachable using IP address 0.0.0.0.\")\n",
    "\n",
    "\n",
    "db = client[\"data-management-db\"]\n",
    "collection_google = db[\"google_data\"]\n",
    "collection_trip = db[\"trip_data\"]\n",
    "\n",
    "# Insert the dataframe into the collection\n",
    "collection_google.insert_many(df.to_dict('records'))\n",
    "collection_trip.insert_many(df2.to_dict('records'))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.2"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "aee8b7b246df8f9039afb4144a1f6fd8d2ca17a180786b69acc140d282b71a49"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
